lib/collector/http-agents.js

```diff
@@ -52,66 +51,34 @@ exports.proxyAgent = function proxyAgent(config) {
   if (null !== agentProxyWithKeepAlive) {
     return agentProxyWithKeepAlive
   }
-  const opts = proxyOptions(config)
-  const proxyUrl = opts.proxy_url
+  const proxyUrl = proxyOptions(config)
 
   const proxyOpts = {
-    host: proxyUrl.host,
-    port: proxyUrl.port,
-    protocol: proxyUrl.protocol,
     secureEndpoint: config.ssl,
     auth: proxyUrl.auth,
-    ca: opts.certificates,
+    ca: config?.certificates?.length ? config.certificates : [],
     keepAlive: true
   }
 
-  logger.info(
-    {
-      host: proxyOpts.host,
-      port: proxyOpts.port,
-      auth: !!proxyOpts.auth,
-      protocol: proxyUrl.protocol
-    },
-    'using proxy'
-  )
-
-  agentProxyWithKeepAlive = new ProxyAgent(proxyOpts)
+  logger.info(`using proxy: ${proxyUrl}`)
+  agentProxyWithKeepAlive = new HttpsProxyAgent(proxyUrl, proxyOpts)
   return agentProxyWithKeepAlive
 }
 
 function proxyOptions(config) {
   let proxyUrl
   if (config.proxy) {
-    const parsedUrl = parse(config.proxy)
-
-    proxyUrl = {
-      protocol: parsedUrl.protocol || 'https:',
-      host: parsedUrl.hostname,
-      port: parsedUrl.port || 80,
-      auth: parsedUrl.auth
-    }
+    proxyUrl = config.proxy
   } else {
+    proxyUrl = 'https://'
     let proxyAuth = config.proxy_user
     if (config.proxy_pass !== '') {
       proxyAuth += ':' + config.proxy_pass
+      proxyUrl += `${proxyAuth}@`
     }
 
-    // Unless a proxy config is provided, default to HTTP.
-    proxyUrl = {
-      protocol: 'https:',
-      host: config.proxy_host || 'localhost',
-      port: config.proxy_port || 80,
-      auth: proxyAuth
-    }
-  }
-
-  const opts = {
-    proxy_url: proxyUrl
-  }
-
-  if (config.certificates && config.certificates.length > 0) {
-    opts.certificates = config.certificates
+    proxyUrl += `${config.proxy_host || 'localhost'}:${config.proxy_port || 80}`
```

Comment:
Agreed that 80 would be the wrong port. Do we want to log a todo?

Reply:
I'm hesitant to change too much here.